### PR TITLE
fix(gatsby-remark-image-contentful): show useful error message for files that can not be rendered as image

### LINKED ISSUE #32511: [gatsby-remark-images-contentful] No stack trace on unsupported image format
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

We've spent a couple of hours debugging our Gatsby builds because of this error: `Error: Input buffer contains unsupported image format`. We had just added `gatsby-remark-images-contentful` and then our builds started failing. It took a couple of hours to figure out that someone has accidentally prefixed a markdown link with a `!`, turning it into an image instead. The url linked to a `.zip` file: `alt here` and caused the complete build to crash with no traces where to start looking for errors.
Impact: several people, including those from Gatsby spending hours to figure out what might be going on. A clear stack trace would have sped up the process quite a lot.

### Reproduction Link

https://github.com/bartveneman/gatsby-repro-contentful-images-failure

### Steps to Reproduce

1. Add an image with a link to an unsupported image type to a markdown input (locally or in Contentful), for example: `alt here`
2. `npm run dev` or `npm run build`
3. The error is right there

### Expected Result

A clear error message telling *which* image type isn't supported, which url it was trying to optimize and maybe even the location of the Markdown the image was in.

### Actual Result

Only a minial error message is shown, without any leads where to investigate further: `Error: Input buffer contains unsupported image format`

### Environment

```shell
System:
    OS: macOS 11.4
    CPU: (8) x64 Intel(R) Core(TM) i5-8259U CPU @ 2.30GHz
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.15.3 - ~/.nvm/versions/node/v14.15.3/bin/node
    Yarn: 1.22.5 - ~/.yarn/bin/yarn
    npm: 7.5.2 - ~/.nvm/versions/node/v14.15.3/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 88.0.4324.150
    Firefox: 89.0.2
    Safari: 14.1.1
  npmPackages:
    gatsby: ^3.10.1 => 3.10.1 
    gatsby-plugin-sharp: ^3.10.2 => 3.10.2 
    gatsby-remark-images-contentful: ^4.7.0 => 4.7.0 
    gatsby-source-filesystem: ^3.10.0 => 3.10.0 
    gatsby-transformer-remark: ^4.7.0 => 4.7.0
```

### Config Flags

N/A

### PR DESCRIPTION
**Pull request by axe312ger**
fixes #32511

### CODE REVIEW

**Comment by bartveneman** [comment]
Tested this locally in my reduced tescase from #32511 and works great!

**Comment by axe312ger** [comment]
@bartveneman awesome, thanks for testing.

File: packages/gatsby-remark-images-contentful/src/__tests__/index.js [hunk 1]
```diff
@@ -16,18 +16,28 @@ jest.mock(`../utils/`, () => {
...
-  return {
-    metadata: jest.fn(() => {
-      return {
-        width: 200,
-        height: 200,
-        density: 75,
-      }
-    }),
-  }
+  return { metadata: mockedSharpMetadata }
```
Comment (wardpeet) [inline packages/gatsby-remark-images-contentful/src/__tests__/index.js:22]:
```suggestion
  return { metadata: jest.fn() }
```

File: packages/gatsby-remark-images-contentful/src/__tests__/index.js [hunk 2]
```diff
@@ -16,18 +16,28 @@ jest.mock(`../utils/`, () => {
...
+      density: 75,
+    }
+  })
+}
+
+const mockSharpFailure = () => {
+  mockedSharpMetadata = jest.fn(() =>
+    Promise.reject(new Error(`invalid image`))
+  )
+}
```
Comment (wardpeet) [inline packages/gatsby-remark-images-contentful/src/__tests__/index.js:39]: This is the proper way to mock :)

```suggestion
const sharp = require('sharp');

const mockSharpSuccess = () => {
  sharp.metadata.mockResolvedValueOnce({
    width: 200,
    height: 200,
    density: 75,
  })
}

const mockSharpFailure = () => {
  sharp.metadata.mockRejectedValue(new Error(`invalid image`)
}
```

**Review by wardpeet — COMMENTED** [review]
small test nits

File: packages/gatsby-remark-images-contentful/src/__tests__/index.js (hunk 1 above)
Comment (axe312ger) [reply packages/gatsby-remark-images-contentful/src/__tests__/index.js]: I changed it closer to how gatsby-plugin-sharp mocks sharp. Please have a look

File: packages/gatsby-remark-images-contentful/src/__tests__/index.js (hunk 2 above)
Comment (axe312ger) [reply packages/gatsby-remark-images-contentful/src/__tests__/index.js]: see comment above

**Review by wardpeet — APPROVED** [review]

**Pull request merged**